src/main/java/com/zaxxer/hikari/pool/PoolEntry.java

```diff
@@ -90,7 +90,9 @@ void setFutureEol(final ScheduledFuture<?> endOfLife)
 
    Connection createProxyConnection(final ProxyLeakTask leakTask, final long now)
    {
-      return ProxyFactory.getProxyConnection(this, connection, openStatements, leakTask, now, hikariPool.isAutoCommit());
+      ProxyConnection pc = ProxyFactory.getProxyConnection(this, connection, openStatements, leakTask, now);
+      hikariPool.initProxyConnection(pc);
```

Comment:
Let me benchmark it, and I'll let you know.  Almost midnight here, so probably tomorrow morning.